gadget/update.go

```diff
@@ -21,12 +21,78 @@ package gadget
 import (
 	"errors"
 	"fmt"
+
+	"github.com/snapcore/snapd/logger"
 )
 
 var (
-	ErrNoUpdate = errors.New("no update needed")
```

Comment:
nitpick: I would use different var groupings for Err* variable and the non-exported ones